### ISSUE TITLE
Ability to pass a dummy 'var' param to procs

### ISSUE THREAD

**Issue by dom96** [nim-lang/Nim#5759]
```nim
proc foo(x: var int) =
  x.inc()


foo(_)
```

I've found a use for something like this a couple of times now.

**Comment by krux02** [comment]
Not sure if this is really justifies as a new language feature that needs to be documented and maintained.  I have two arguments why this should not be a language feature:

* It's not really hard to write a workaround:
```Nim
var unused1: int
foo(unused1)
```
* For me having to used an explicit _unused_ parameter is a good indication, that the code can be written in a better way. E.g. don't use var parameters, but properly split input and output and avoid side effects:
```Nim
proc foo(x: int): int = x + 1
discard foo(0)
```

**Comment by andreaferretti** [comment]
I agree with @krux02 - it would be yet another language level feature, for no clear gain

**Comment by Araq** [comment]
Untested:

```nim
template witness(T: typedesc): untyped = (var x: T; x)

proc foo(x: var int) =
  x.inc()

foo(witness(int))
```

**Comment by Varriount** [comment]
I also agree with @krux02
